frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -189,44 +211,152 @@ export const getEventSourceData = (source: string) => {
+          referenceFor(res) === referenceForModel(EventSourceCamelModel) &&
+          res?.metadata?.labels?.[IS_CAMEL_CONNECTOR_SOURCE] === 'true'
+        ) {
+          const { apiGroup, plural, kind } = modelFor(referenceFor(res));
+          const {
+            metadata: { name, annotations },
+          } = res;
+          const modelData = {
+            [name]: {
+              name,
+              iconUrl: isIconUrl(annotations?.[EVENT_SOURCE_ICON])
+                ? annotations?.[EVENT_SOURCE_ICON]
+                : getKnativeEventSourceIcon(kind),
+              displayName: _.startCase(name),
+              title: _.startCase(name),
+              data: { almData: res },
+            },
+          };
+          acm.push(
+            checkAccess({
+              group: apiGroup,
+              resource: plural,
+              namespace,
+              verb: 'create',
+            }).then((result) => (result.status.allowed ? modelData : {})),
+          );
+        }
+      });
+      return acm;
+    },
+    [],
+  );
+
+export const getEventSourceList = (namespace: string, eventSourceModels: K8sKind[]) => {
+  const accessList = [];
+  eventSourceModels.map((model) => {
+    const { apiGroup, plural, kind } = model;
+    const modelData = {
+      [kind]: {
+        name: kind,
+        iconUrl: getKnativeEventSourceIcon(kind),
+        displayName: _.startCase(kind),
+        title: _.startCase(kind),
+      },
+    };
+    return accessList.push(
+      checkAccess({
+        group: apiGroup,
+        resource: plural,
+        namespace,
+        verb: 'create',
+      }).then((result) => (result.status.allowed ? modelData : {})),
+    );
+  });
+  return accessList;
+};
+
+// To order sources with known followed by CamelSource, followed by camelConnector and other dynamic sources
+export const sortSourcesData = (sourcesObj: NormalizedEventSources): NormalizedEventSources => {
+  const sortSourcesList: EventSourceList[] = _.orderBy(
+    Object.values(sourcesObj),
+    ['name'],
+    ['asc'],
+  );
+  const knownSourcesKind = [
+    EventSourceApiServerModel.kind,
+    EventSourceContainerModel.kind,
+    EventSourceCronJobModel.kind,
+    EventSourceKafkaModel.kind,
+    EventSourcePingModel.kind,
+    EventSourceSinkBindingModel.kind,
+  ];
+  const knownSourcesList = _.filter(sortSourcesList, (source) =>
+    knownSourcesKind.includes(source.name),
+  );
+  const camelSourcesList = _.filter(
+    sortSourcesList,
+    (source) => EventSourceCamelModel.kind === source?.name,
+  );
+  const dynamicCamelConnectorsList = _.filter(
+    sortSourcesList,
+    (source) =>
+      !knownSourcesKind.includes(source.name) &&
+      source.name !== EventSourceCamelModel.kind &&
+      !!source?.data,
+  );
+  const dynamicSourcesList = _.filter(
+    sortSourcesList,
+    (source) =>
+      !knownSourcesKind.includes(source.name) &&
+      source.name !== EventSourceCamelModel.kind &&
+      !source?.data,
```

Comment:
Sometimes you have `source?.` while other times you have `source.`. If `source` can be undefined, always use `?`

Reply:
Updated it , source won't be undefined here as iterating in filter